### PR TITLE
Coronavirus Research - add height and weight widgets

### PR DESCRIPTION
**Pull request by stephenBarrs**
## Description
Add and enhance styling of Height and Weight widgets for forms. Product design requests height and weight inputs to have a label adjacent and to the right of the input field so custom widgets were created. 

Add unit tests for form elements

## Testing done
Manual testing, unit tests

## Screenshots
image
image

## Acceptance criteria
- [ ] Height widget display feet and inches inputs with labels on a single line
- [ ] Weight widget displays input and label on single line 
- [ ] All unit tests pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/coronavirus-research/widgets/HeightWidget.jsx [hunk 1]
```diff
@@ -13,18 +13,24 @@ function getEmptyState() {
...
   };
 }
-
+function setLabelStyles(newState) {
+  let labelMarginLeftClass = 'vads-u-margin-left--1';
+  if (
+    newState &&
+    newState.touched.feet &&
+    newState.touched.inches &&
+    (!newState.value.feet || !newState.value.inches)
```
Comment (mchelen-gov) [inline src/applications/coronavirus-research/widgets/HeightWidget.jsx:22]: can this be simplified with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining ?

File: src/applications/coronavirus-research/widgets/HeightWidget.jsx [hunk 2]
```diff
@@ -13,18 +13,24 @@ function getEmptyState() {
...
+  ) {
+    labelMarginLeftClass = 'vads-u-margin-left--3';
+  }
+  return `vads-l-col--1 vads-u-margin-top--3 ${labelMarginLeftClass}`;
+}
 const calculateHeight = ({ feet, inches }) =>
```
Comment (mchelen-gov) [inline src/applications/coronavirus-research/widgets/HeightWidget.jsx:28]: is this intended to return a height in decimal value?

File: src/applications/coronavirus-research/widgets/HeightWidget.jsx [hunk 3]
```diff
@@ -36,61 +42,58 @@ export default class HeightWidget extends React.Component {
...
   handleChange = (field, value) => {
     let newState = _.set(['value', field], value, this.state);
     newState = _.set(['touched', field], true, newState);
-
     this.setState(newState, () => {
       if (this.props.required && this.isIncomplete(newState.value)) {
         this.props.onChange();
       } else {
         this.props.onChange(calculateHeight(newState.value));
       }
```
Comment (mchelen-gov) [inline src/applications/coronavirus-research/widgets/HeightWidget.jsx:57]:
```suggestion
      const height =
        this.props.required && this.isIncomplete(newState.value)
          ? calculateHeight(newState.value)
          : undefined;
      this.props.onChange(height);
      this.labelStyleClasses = setLabelStyles(newState);
```
does this work?

File: src/applications/coronavirus-research/widgets/HeightWidget.jsx (hunk 2 above)
Comment (stephenBarrs) [reply src/applications/coronavirus-research/widgets/HeightWidget.jsx:28]: @mchelen-gov  This does not and should not return a decimal value.  This translates the value of height in feet and inches to height in inches

File: src/applications/coronavirus-research/widgets/HeightWidget.jsx (hunk 3 above)
Comment (stephenBarrs) [reply src/applications/coronavirus-research/widgets/HeightWidget.jsx:57]: Ternary is better way for sure, just had to reverse it:
`const height =
        this.props.required && this.isIncomplete(newState.value)
          ? undefined
          : calculateHeight(newState.value);
      this.props.onChange(height);
      this.labelStyleClasses = setLabelStyles(newState);`

File: src/applications/coronavirus-research/widgets/HeightWidget.jsx [hunk 4]
```diff
@@ -36,61 +42,58 @@ export default class HeightWidget extends React.Component {
       if (this.isTouched(newState.touched)) {
         this.props.onBlur(this.props.id);
       }
+      this.labelStyleClasses = setLabelStyles(newState);
     });
   };
 
   handleChange = (field, value) => {
     let newState = _.set(['value', field], value, this.state);
     newState = _.set(['touched', field], true, newState);
```
[the remark this answers is not in the document]
Comment (stephenBarrs) [reply src/applications/coronavirus-research/widgets/HeightWidget.jsx:51]: @mchelen-gov one is setting `value` and one is setting `touched`.

File: src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js [hunk 5]
```diff
@@ -5,7 +5,6 @@ import {
   personDescriptionText,
 } from '../content/individualsInvolved';
 import IndividualsInvolvedCard from '../components/IndividualsInvolvedCard';
-import fullNameUI from 'platform/forms/definitions/fullName';
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js:8]: I'm not sure why this was removed? It appears to still be in use in the `covidResearchUISchema` above.

Comment (stephenBarrs) [reply src/applications/disability-benefits/all-claims/pages/individualsInvolvedFollowUp.js:8]: Long slack thread here: https://dsva.slack.com/archives/CBU0KDSB1/p1598458760323500


TL;DR: individualsInvolvedFollowUp.js was using `fullNameUI` (which includes suffix) for the UISchema but a custom element for Schema.  This has not been an issue until the unit tests for `coronavirus-screener` were introduced with this PR.  For some reason, the tests in [this file ](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/disability-benefits/all-claims/tests/pages/individualsInvolvedFollowUp.unit.spec.jsx) began failing due to error: `Error: uiSchema order list contains extraneous property 'suffix'.`.  I removed `fullNameUI` and replaced with a custom element in the UISchema to see if all test would pass in Circle. 

I plan to seek out an engineer more familiar with `individualsInvolvedFollowUp.js` and ask them to review this as well.  

@mchelen-gov and @billfienberg  are also helping analyze this.

**Review by micahchiang — APPROVED** [review]
Not sure why the aforementioned unit test issue cropped up, but I can't reproduce, and the most recent commit seems to fix it. PR looks good otherwise.

**Review by batemapf — APPROVED** [review]
